Selection API: Return live range synchronized with selection from getRangeAt and throw errors as specified

rniwa opened the pull request:
#### 1226d38de5c750fc736ce7075e86e248674ee430
<pre>
Selection API: Return live range synchronized with selection from getRangeAt and throw errors as specified
<a href="https://bugs.webkit.org/show_bug.cgi?id=216325">https://bugs.webkit.org/show_bug.cgi?id=216325</a>

Reviewed by Darin Adler.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml: Enabled the feature.
* LayoutTests/editing/deleting/backspace-delete-last-char-in-table-expected.txt: Initial selection has changed.
* LayoutTests/editing/deleting/delete-4038408-fix-expected.txt: Ditto.
* LayoutTests/editing/deleting/delete-image-004-expected.txt: Editing delegate callback change.
* LayoutTests/editing/deleting/delete-list-items-in-table-cell-1-expected.txt: Initial selection has changed.
* LayoutTests/editing/deleting/delete-list-items-in-table-cell-2-expected.txt: Ditto.
* LayoutTests/editing/deleting/delete-list-items-in-table-cell-5-expected.txt: Ditto.
* LayoutTests/editing/deleting/delete-list-items-in-table-cell-7-expected.txt: Ditto.
* LayoutTests/editing/deleting/delete-start-block-expected.txt: Rebaselined. WebKit still does not hit any assertions or crash.
* LayoutTests/editing/deleting/deleting-relative-positioned-special-element-expected.txt: Initial selection has changed.
* LayoutTests/editing/deleting/forward-delete-empty-table-cell-expected.txt: Ditto.
* LayoutTests/editing/deleting/forward-delete-last-char-in-table-expected.txt: Ditto.
* LayoutTests/editing/deleting/merge-paragraph-contatining-noneditable-expected.txt: Ditto.
* LayoutTests/editing/editability/ignored-content-expected.txt: Visually identical selection.
* LayoutTests/editing/execCommand/button-expected.txt: Ditto.
* LayoutTests/editing/execCommand/change-list-type-expected.txt: Initial selection has changed.
* LayoutTests/editing/execCommand/contenteditable-justify-next-paragraph-expected.txt: Final selection has changed but it&apos;s visually identical.
* LayoutTests/editing/execCommand/format-block-multiple-paragraphs-expected.txt: Initial selection has changed.
* LayoutTests/editing/execCommand/format-block-multiple-paragraphs-in-pre-expected.txt: Ditto.
* LayoutTests/editing/execCommand/format-block-table-expected.txt: Ditto.
* LayoutTests/editing/execCommand/format-block-with-braces-expected.txt: Editing delegate callback change.
* LayoutTests/editing/execCommand/indent-pre-list-expected.txt: Initial selection has changed.
* LayoutTests/editing/execCommand/indent-pre-paragraphs-expected.txt: Ditto.
* LayoutTests/editing/execCommand/insert-nested-lists-in-table-expected.txt: Visually identical selection.
* LayoutTests/editing/execCommand/insert-paragraph-twice-at-end-of-block-styled-listitem-expected.txt: Initial selection has changed.
* LayoutTests/editing/execCommand/insert-paragraph-twice-at-end-of-custom-listitem-expected.txt: Ditto.
* LayoutTests/editing/execCommand/insertHorizontalRule-expected.txt: Progression. Caret is between &quot;foo&quot; and &quot;bar&quot; as expected.
* LayoutTests/editing/execCommand/unindent-nested-blockquote-with-inner-div-expected.txt: Initial selection has changed.
* LayoutTests/editing/inserting/insert-list-in-table-cell-01-expected.txt: Ditto.
* LayoutTests/editing/inserting/insert-list-in-table-cell-02-expected.txt: Ditto.
* LayoutTests/editing/inserting/insert-list-in-table-cell-03-expected.txt: Ditto.
* LayoutTests/editing/inserting/insert-list-in-table-cell-04-expected.txt: Ditto.
* LayoutTests/editing/inserting/insert-list-in-table-cell-05-expected.txt: Ditto.
* LayoutTests/editing/inserting/insert-list-in-table-cell-06-expected.txt: Ditto.
* LayoutTests/editing/inserting/insert-list-in-table-cell-07-expected.txt: Ditto.
* LayoutTests/editing/inserting/insert-list-in-table-cell-08-expected.txt: Ditto.
* LayoutTests/editing/pasteboard/cleanup-on-move-expected.txt: Visually identical selection.
* LayoutTests/editing/pasteboard/copy-paste-with-important-rules-expected.txt: Initial selection has changed.
* LayoutTests/editing/pasteboard/copy-text-with-backgroundcolor-expected.txt: Ditto.
* LayoutTests/editing/pasteboard/display-block-on-spans-expected.txt: Ditto.
* LayoutTests/editing/pasteboard/drag-drop-url-with-style-expected.txt: Visually identical selection.
* LayoutTests/editing/pasteboard/drag-list-item-expected.txt: Ditto.
* LayoutTests/editing/pasteboard/paste-table-003-expected.txt: Editing delegate callback change.
* LayoutTests/editing/pasteboard/paste-table-cells-expected.txt: Initial selection has changed.
* LayoutTests/editing/pasteboard/paste-table-with-unrendered-text-nodes-expected.txt: Ditto.
* LayoutTests/editing/pasteboard/paste-text-005-expected.txt: Visually identical selection.
* LayoutTests/editing/pasteboard/paste-text-006-expected.txt: Ditto.
* LayoutTests/editing/pasteboard/paste-text-007-expected.txt: Ditto.
* LayoutTests/editing/pasteboard/paste-text-with-style-expected.txt: Initial selection has changed.
* LayoutTests/editing/pasteboard/select-element-1-expected.txt: Editing delegate callback change.
* LayoutTests/editing/pasteboard/testcase-9507-expected.txt: Initial selection has changed.
* LayoutTests/editing/selection/containsNode-expected.txt: Rebaselined.
* LayoutTests/editing/selection/containsNode.html: Changed the expectation for partially selected text node when
partialContainment flag is not set. When live range was disabled, we&apos;d always behave as if partial containment is set to true.
* LayoutTests/editing/selection/dump-as-markup-expected.txt: Initial selection has changed.
* LayoutTests/editing/selection/editable-table-cell-selection.html: Updated the expected result for selecting a single cell
since we no longer include the new line at the end.
* LayoutTests/editing/selection/extend-after-mouse-selection-expected.txt: Visually identical selection.
* LayoutTests/editing/selection/extend-selection-backward-at-beginning-of-line-by-sentence-granularity-expected.txt: Ditto.
* LayoutTests/editing/selection/extend-selection-backward-at-beginning-of-line-by-word-granularity-expected.txt: Ditto.
* LayoutTests/editing/selection/focus-contenteditable-iframe-expected.txt: Initial selection has changed.
* LayoutTests/editing/selection/getRangeAt-expected.txt: Visually identical selection.
* LayoutTests/editing/selection/move-by-word-visually-across-object-element-1-expected.txt: Initial selection has changed.
* LayoutTests/editing/selection/move-by-word-visually-across-object-element-2-expected.txt: Ditto.
* LayoutTests/editing/selection/move-by-word-visually-across-object-element-3-expected.txt: Ditto.
* LayoutTests/editing/selection/node-removal-1-expected.txt: Visually identical selection.
* LayoutTests/editing/selection/select-out-of-floated-non-editable-02-expected.txt: Ditto.
* LayoutTests/editing/selection/select-out-of-floated-non-editable-07-expected.txt: Ditto.
* LayoutTests/editing/selection/select-out-of-floated-non-editable-10-expected.txt: Ditto.
* LayoutTests/editing/selection/shift-click-includes-existing-selection-expected.txt: Initial selection has changed.
* LayoutTests/editing/selection/user-select-all-selection-expected.txt: Visually identical selection.
* LayoutTests/editing/selection/user-select-all-with-shift-expected.txt: Ditto.
* LayoutTests/editing/selection/user-select-all-with-single-click-expected.txt: Ditto.
* LayoutTests/editing/selection/user-select-js-property.html: Changed the expectation of the first test case. New expected result matches
Chrome and Firefox.
* LayoutTests/editing/style/change-text-direction-crash-expected.txt: Visually identical selection.
* LayoutTests/editing/style/fore-color-by-name-expected.txt: Visually identical selection.
* LayoutTests/editing/style/preserve-selection-direction-expected.txt: Ditto.
* LayoutTests/editing/style/remove-underline-from-stylesheet-expected.txt: Ditto.
* LayoutTests/editing/style/style-boundary-001-expected.txt: Ditto.
* LayoutTests/editing/style/style-text-node-without-editable-parent-expected.txt: Ditto.
* LayoutTests/editing/undo/redo-style-expected.txt: Ditto.
* LayoutTests/editing/undo/remove-css-property-and-remove-style-expected.txt: Ditto.
* LayoutTests/fast/attachment/attachment-select-on-click-expected.txt: Ditto.
* LayoutTests/fast/dom/delete-contents-expected.txt: Rebaselined. Progression. New behaivor matches Chrome and Firefox.
* LayoutTests/fast/editing/editing-position-crash.html: Fixed the test to use a valid offset.
* LayoutTests/fast/events/before-input-events-prevent-inline-text-direction-expected.txt: Visually identical selection.
* LayoutTests/fast/rendering/render-multi-column-set-invalidate-crash.html: Fixed the test not to use invalid offsets.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/email-set-value-expected.txt: Rebaselined.
New behavior matches that of Firefox.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/selection-pointer-expected.txt: Rebaselined.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-00-expected.txt: Rebaselined now that all test cases are passing
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-04-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-12-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-16-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-20-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-24-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-28-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-32-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-36-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange.tentative-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/collapse-00-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/collapse-15-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/collapse-45-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/collapse-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/collapseToStartEnd-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/deleteFromDocument-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/extend-00-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/extend-20-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/extend-40-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/isCollapsed-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/removeRange-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/selectAllChildren-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/setBaseAndExtent-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/type-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/deleting/delete-image-001-expected.txt: Editing delegate callback change.
* LayoutTests/platform/gtk/editing/deleting/delete-image-002-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/deleting/delete-image-003-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/deleting/delete-image-004-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/deleting/delete-line-001-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/deleting/delete-line-003-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/deleting/delete-line-005-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/style/create-block-for-style-001-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/style/create-block-for-style-004-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/style/create-block-for-style-007-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/style/preserve-selection-direction-expected.txt: Added.
* LayoutTests/platform/gtk/editing/unsupported-content/list-delete-001-expected.txt: Editing delegate callback change.
* LayoutTests/platform/gtk/editing/unsupported-content/list-delete-003-expected.txt: Ditto.
* LayoutTests/platform/ios-wk2/editing/style/create-block-for-style-007-expected.txt: Ditto.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/selection-pointer-expected.txt: Rebaselined.
* LayoutTests/platform/ios/editing/deleting/delete-image-001-expected.txt: Editing delegate callback change.
* LayoutTests/platform/ios/editing/deleting/delete-image-002-expected.txt: Ditto.
* LayoutTests/platform/ios/editing/deleting/delete-image-003-expected.txt: Ditto.
* LayoutTests/platform/ios/editing/deleting/delete-image-004-expected.txt: Ditto.
* LayoutTests/platform/ios/editing/deleting/delete-line-001-expected.txt: Ditto.
* LayoutTests/platform/ios/editing/deleting/delete-line-003-expected.txt: Ditto.
* LayoutTests/platform/ios/editing/deleting/delete-line-005-expected.txt: Ditto.
* LayoutTests/platform/ios/editing/style/create-block-for-style-001-expected.txt: Ditto.
* LayoutTests/platform/ios/editing/style/create-block-for-style-004-expected.txt: Ditto.
* LayoutTests/platform/ios/editing/unsupported-content/list-delete-003-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/deleting/delete-image-001-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/deleting/delete-image-002-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/deleting/delete-image-003-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/deleting/delete-line-001-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/deleting/delete-line-003-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/deleting/delete-line-005-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/style/create-block-for-style-001-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/style/create-block-for-style-004-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/style/create-block-for-style-007-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/unsupported-content/list-delete-001-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/unsupported-content/list-delete-003-expected.txt: Ditto.
* LayoutTests/platform/win/editing/execCommand/button-expected.txt: Visually identical selection.
* LayoutTests/platform/win/editing/execCommand/insert-list-nested-with-orphaned-live-range-expected.txt: Rebaselined.
* LayoutTests/platform/win/editing/style/preserve-selection-direction-expected.txt: Added.
* LayoutTests/platform/win/editing/style/remove-underline-from-stylesheet-expected.txt: Rebaselined.
* LayoutTests/platform/win/fast/forms/shadow-tree-exposure-expected.txt: Added.
* Tools/TestWebKitAPI/Tests/WebKit/simple-image-overlay.html: Fixed the test.

Canonical link: <a href="https://commits.webkit.org/259904@main">https://commits.webkit.org/259904@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/26a0499d506bd92ad8bfd447e749b126f2a49713

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/105372 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/14442 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38247 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/114626 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/174804 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/109273 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/15629 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/5376 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97677 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/114521 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111130 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/12084 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95066 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/39565 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/93950 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/26700 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/81242 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/95169 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/7762 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28060 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/93267 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/5520 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/7877 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/4648 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/30352 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/13905 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/47604 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/101971 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/9667 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/25431 "Passed tests") | 
| [❌ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3688 "Failed to update pull request") | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on LayoutTests/fast/editing/editing-position-crash.html:
```diff
@@ -21,7 +21,7 @@
 <body>
 <svg onfocusin="insert()">
-<path id="path0" tabindex="2" onfocusin="window.getSelection().setPosition(obj0, 10);" />
+<path id="path0" tabindex="2" onfocusin="window.getSelection().setPosition(obj0, 1);" />
```
Surprised you did not land this change separately

rniwa replied:
Yeah, I noticed I forgot to split 3-4 test changes after updating the change log. We could split into their own PRs if you'd like.

darinadler replied:
Does not seem necessary if this is ready to go. If we end up having to revert this for some reason would be nice to have it be as small as possible.

darinadler commented on LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/email-set-value-expected.txt:
```diff
@@ -1,4 +1,4 @@
 
 
-PASS setValue(sanitizedValue) is reflected in visible text field content
+FAIL setValue(sanitizedValue) is reflected in visible text field content assert_equals: visible value is unsanitized expected " foo@bar   " but got ""
```
Should we be changing the test expectation here? You say the new behavior matches Firefox. Why leave WPT expecting the wrong thing? Should the test be more lenient? Or should it be fixed?

darinadler replied:
Also note the change log has the misspelling “Rebsaelined” a few times.

rniwa replied:
> Should we be changing the test expectation here? You say the new behavior matches Firefox. Why leave WPT expecting the wrong thing? Should the test be more lenient? Or should it be fixed?

Yeah, I have a WPT issue filed to track this:
https://github.com/web-platform-tests/wpt/issues/37214

There is a bit of chicken & egg issue here in that right now 2 out of 3 engines are aligned to include the offending text. Once this patch lands (and sticks), that'll change. At that point, we can definitely update WPT to expect the new behavior.

darinadler replied:
It seems like we can fix the test to not depend on this old WebKit quirk without waiting for engines to change. I don’t think the test is intended to test what `getSelection().toString()` does. I suppose it’s OK to have a test for that, but this test is supposed to check how the sanitizing of that element works, which we should find another way to test, right?

rniwa replied:
The test is checking what the visible value of the text is (it's not sanitized). I'm not aware of any other way to do that so "fixing" the test would basically mean getting rid of these checks.

darinadler replied:
Maybe copy and paste sees this “visible” value?

rniwa replied:
I guess one way to test it is turn it into a ref test.

rniwa replied:
Alright, posted a WPT PR here to convert it to a ref test: https://github.com/web-platform-tests/wpt/pull/37773

rniwa committed:
Selection API: Return live range synchronized with selection from getRangeAt and throw errors as specified
https://bugs.webkit.org/show_bug.cgi?id=216325

Reviewed by Darin Adler.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml: Enabled the feature.
* LayoutTests/editing/deleting/backspace-delete-last-char-in-table-expected.txt: Initial selection has changed.
* LayoutTests/editing/deleting/delete-4038408-fix-expected.txt: Ditto.
* LayoutTests/editing/deleting/delete-image-004-expected.txt: Editing delegate callback change.
* LayoutTests/editing/deleting/delete-list-items-in-table-cell-1-expected.txt: Initial selection has changed.
* LayoutTests/editing/deleting/delete-list-items-in-table-cell-2-expected.txt: Ditto.
* LayoutTests/editing/deleting/delete-list-items-in-table-cell-5-expected.txt: Ditto.
* LayoutTests/editing/deleting/delete-list-items-in-table-cell-7-expected.txt: Ditto.
* LayoutTests/editing/deleting/delete-start-block-expected.txt: Rebaselined. WebKit still does not hit any assertions or crash.
* LayoutTests/editing/deleting/deleting-relative-positioned-special-element-expected.txt: Initial selection has changed.
* LayoutTests/editing/deleting/forward-delete-empty-table-cell-expected.txt: Ditto.
* LayoutTests/editing/deleting/forward-delete-last-char-in-table-expected.txt: Ditto.
* LayoutTests/editing/deleting/merge-paragraph-contatining-noneditable-expected.txt: Ditto.
* LayoutTests/editing/editability/ignored-content-expected.txt: Visually identical selection.
* LayoutTests/editing/execCommand/button-expected.txt: Ditto.
* LayoutTests/editing/execCommand/change-list-type-expected.txt: Initial selection has changed.
* LayoutTests/editing/execCommand/contenteditable-justify-next-paragraph-expected.txt: Final selection has changed but it's visually identical.
* LayoutTests/editing/execCommand/format-block-multiple-paragraphs-expected.txt: Initial selection has changed.
* LayoutTests/editing/execCommand/format-block-multiple-paragraphs-in-pre-expected.txt: Ditto.
* LayoutTests/editing/execCommand/format-block-table-expected.txt: Ditto.
* LayoutTests/editing/execCommand/format-block-with-braces-expected.txt: Editing delegate callback change.
* LayoutTests/editing/execCommand/indent-pre-list-expected.txt: Initial selection has changed.
* LayoutTests/editing/execCommand/indent-pre-paragraphs-expected.txt: Ditto.
* LayoutTests/editing/execCommand/insert-nested-lists-in-table-expected.txt: Visually identical selection.
* LayoutTests/editing/execCommand/insert-paragraph-twice-at-end-of-block-styled-listitem-expected.txt: Initial selection has changed.
* LayoutTests/editing/execCommand/insert-paragraph-twice-at-end-of-custom-listitem-expected.txt: Ditto.
* LayoutTests/editing/execCommand/insertHorizontalRule-expected.txt: Progression. Caret is between "foo" and "bar" as expected.
* LayoutTests/editing/execCommand/unindent-nested-blockquote-with-inner-div-expected.txt: Initial selection has changed.
* LayoutTests/editing/inserting/insert-list-in-table-cell-01-expected.txt: Ditto.
* LayoutTests/editing/inserting/insert-list-in-table-cell-02-expected.txt: Ditto.
* LayoutTests/editing/inserting/insert-list-in-table-cell-03-expected.txt: Ditto.
* LayoutTests/editing/inserting/insert-list-in-table-cell-04-expected.txt: Ditto.
* LayoutTests/editing/inserting/insert-list-in-table-cell-05-expected.txt: Ditto.
* LayoutTests/editing/inserting/insert-list-in-table-cell-06-expected.txt: Ditto.
* LayoutTests/editing/inserting/insert-list-in-table-cell-07-expected.txt: Ditto.
* LayoutTests/editing/inserting/insert-list-in-table-cell-08-expected.txt: Ditto.
* LayoutTests/editing/pasteboard/cleanup-on-move-expected.txt: Visually identical selection.
* LayoutTests/editing/pasteboard/copy-paste-with-important-rules-expected.txt: Initial selection has changed.
* LayoutTests/editing/pasteboard/copy-text-with-backgroundcolor-expected.txt: Ditto.
* LayoutTests/editing/pasteboard/display-block-on-spans-expected.txt: Ditto.
* LayoutTests/editing/pasteboard/drag-drop-url-with-style-expected.txt: Visually identical selection.
* LayoutTests/editing/pasteboard/drag-list-item-expected.txt: Ditto.
* LayoutTests/editing/pasteboard/paste-table-003-expected.txt: Editing delegate callback change.
* LayoutTests/editing/pasteboard/paste-table-cells-expected.txt: Initial selection has changed.
* LayoutTests/editing/pasteboard/paste-table-with-unrendered-text-nodes-expected.txt: Ditto.
* LayoutTests/editing/pasteboard/paste-text-005-expected.txt: Visually identical selection.
* LayoutTests/editing/pasteboard/paste-text-006-expected.txt: Ditto.
* LayoutTests/editing/pasteboard/paste-text-007-expected.txt: Ditto.
* LayoutTests/editing/pasteboard/paste-text-with-style-expected.txt: Initial selection has changed.
* LayoutTests/editing/pasteboard/select-element-1-expected.txt: Editing delegate callback change.
* LayoutTests/editing/pasteboard/testcase-9507-expected.txt: Initial selection has changed.
* LayoutTests/editing/selection/containsNode-expected.txt: Rebaselined.
* LayoutTests/editing/selection/containsNode.html: Changed the expectation for partially selected text node when
partialContainment flag is not set. When live range was disabled, we'd always behave as if partial containment is set to true.
* LayoutTests/editing/selection/dump-as-markup-expected.txt: Initial selection has changed.
* LayoutTests/editing/selection/editable-table-cell-selection.html: Updated the expected result for selecting a single cell
since we no longer include the new line at the end.
* LayoutTests/editing/selection/extend-after-mouse-selection-expected.txt: Visually identical selection.
* LayoutTests/editing/selection/extend-selection-backward-at-beginning-of-line-by-sentence-granularity-expected.txt: Ditto.
* LayoutTests/editing/selection/extend-selection-backward-at-beginning-of-line-by-word-granularity-expected.txt: Ditto.
* LayoutTests/editing/selection/focus-contenteditable-iframe-expected.txt: Initial selection has changed.
* LayoutTests/editing/selection/getRangeAt-expected.txt: Visually identical selection.
* LayoutTests/editing/selection/move-by-word-visually-across-object-element-1-expected.txt: Initial selection has changed.
* LayoutTests/editing/selection/move-by-word-visually-across-object-element-2-expected.txt: Ditto.
* LayoutTests/editing/selection/move-by-word-visually-across-object-element-3-expected.txt: Ditto.
* LayoutTests/editing/selection/node-removal-1-expected.txt: Visually identical selection.
* LayoutTests/editing/selection/select-out-of-floated-non-editable-02-expected.txt: Ditto.
* LayoutTests/editing/selection/select-out-of-floated-non-editable-07-expected.txt: Ditto.
* LayoutTests/editing/selection/select-out-of-floated-non-editable-10-expected.txt: Ditto.
* LayoutTests/editing/selection/shift-click-includes-existing-selection-expected.txt: Initial selection has changed.
* LayoutTests/editing/selection/user-select-all-selection-expected.txt: Visually identical selection.
* LayoutTests/editing/selection/user-select-all-with-shift-expected.txt: Ditto.
* LayoutTests/editing/selection/user-select-all-with-single-click-expected.txt: Ditto.
* LayoutTests/editing/selection/user-select-js-property.html: Changed the expectation of the first test case. New expected result matches
Chrome and Firefox.
* LayoutTests/editing/style/change-text-direction-crash-expected.txt: Visually identical selection.
* LayoutTests/editing/style/fore-color-by-name-expected.txt: Visually identical selection.
* LayoutTests/editing/style/preserve-selection-direction-expected.txt: Ditto.
* LayoutTests/editing/style/remove-underline-from-stylesheet-expected.txt: Ditto.
* LayoutTests/editing/style/style-boundary-001-expected.txt: Ditto.
* LayoutTests/editing/style/style-text-node-without-editable-parent-expected.txt: Ditto.
* LayoutTests/editing/undo/redo-style-expected.txt: Ditto.
* LayoutTests/editing/undo/remove-css-property-and-remove-style-expected.txt: Ditto.
* LayoutTests/fast/attachment/attachment-select-on-click-expected.txt: Ditto.
* LayoutTests/fast/dom/delete-contents-expected.txt: Rebaselined. Progression. New behaivor matches Chrome and Firefox.
* LayoutTests/fast/editing/editing-position-crash.html: Fixed the test to use a valid offset.
* LayoutTests/fast/events/before-input-events-prevent-inline-text-direction-expected.txt: Visually identical selection.
* LayoutTests/fast/rendering/render-multi-column-set-invalidate-crash.html: Fixed the test not to use invalid offsets.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/email-set-value-expected.txt: Rebaselined.
New behavior matches that of Firefox.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/selection-pointer-expected.txt: Rebaselined.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-00-expected.txt: Rebaselined now that all test cases are passing
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-04-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-12-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-16-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-20-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-24-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-28-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-32-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-36-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/addRange.tentative-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/collapse-00-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/collapse-15-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/collapse-45-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/collapse-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/collapseToStartEnd-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/deleteFromDocument-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/extend-00-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/extend-20-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/extend-40-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/isCollapsed-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/removeRange-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/selectAllChildren-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/setBaseAndExtent-expected.txt: Ditto.
* LayoutTests/imported/w3c/web-platform-tests/selection/type-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/deleting/delete-image-001-expected.txt: Editing delegate callback change.
* LayoutTests/platform/gtk/editing/deleting/delete-image-002-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/deleting/delete-image-003-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/deleting/delete-image-004-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/deleting/delete-line-001-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/deleting/delete-line-003-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/deleting/delete-line-005-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/style/create-block-for-style-001-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/style/create-block-for-style-004-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/style/create-block-for-style-007-expected.txt: Ditto.
* LayoutTests/platform/gtk/editing/style/preserve-selection-direction-expected.txt: Added.
* LayoutTests/platform/gtk/editing/unsupported-content/list-delete-001-expected.txt: Editing delegate callback change.
* LayoutTests/platform/gtk/editing/unsupported-content/list-delete-003-expected.txt: Ditto.
* LayoutTests/platform/ios-wk2/editing/style/create-block-for-style-007-expected.txt: Ditto.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/selection-pointer-expected.txt: Rebaselined.
* LayoutTests/platform/ios/editing/deleting/delete-image-001-expected.txt: Editing delegate callback change.
* LayoutTests/platform/ios/editing/deleting/delete-image-002-expected.txt: Ditto.
* LayoutTests/platform/ios/editing/deleting/delete-image-003-expected.txt: Ditto.
* LayoutTests/platform/ios/editing/deleting/delete-image-004-expected.txt: Ditto.
* LayoutTests/platform/ios/editing/deleting/delete-line-001-expected.txt: Ditto.
* LayoutTests/platform/ios/editing/deleting/delete-line-003-expected.txt: Ditto.
* LayoutTests/platform/ios/editing/deleting/delete-line-005-expected.txt: Ditto.
* LayoutTests/platform/ios/editing/style/create-block-for-style-001-expected.txt: Ditto.
* LayoutTests/platform/ios/editing/style/create-block-for-style-004-expected.txt: Ditto.
* LayoutTests/platform/ios/editing/unsupported-content/list-delete-003-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/deleting/delete-image-001-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/deleting/delete-image-002-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/deleting/delete-image-003-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/deleting/delete-line-001-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/deleting/delete-line-003-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/deleting/delete-line-005-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/style/create-block-for-style-001-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/style/create-block-for-style-004-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/style/create-block-for-style-007-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/unsupported-content/list-delete-001-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/unsupported-content/list-delete-003-expected.txt: Ditto.
* LayoutTests/platform/win/editing/execCommand/button-expected.txt: Visually identical selection.
* LayoutTests/platform/win/editing/execCommand/insert-list-nested-with-orphaned-live-range-expected.txt: Rebaselined.
* LayoutTests/platform/win/editing/style/preserve-selection-direction-expected.txt: Added.
* LayoutTests/platform/win/editing/style/remove-underline-from-stylesheet-expected.txt: Rebaselined.
* LayoutTests/platform/win/fast/forms/shadow-tree-exposure-expected.txt: Added.
* Tools/TestWebKitAPI/Tests/WebKit/simple-image-overlay.html: Fixed the test.

Canonical link: https://commits.webkit.org/259904@main
